### ISSUE TITLE
HXDEV repo location

### ISSUE THREAD

**Issue by dajhorn** [Baron-von-Riedesel/HX#44]
Does the [HXDEV221.ZIP](https://github.com/Baron-von-Riedesel/HX/releases/tag/v2.21) package have a git repository?

I want to submit a PR for the `OWSUPP/README.TXT` file for Open Watcom 1.9 and v2.

**Comment by Baron-von-Riedesel** [comment]
> Does the HXDEV221.ZIP package have a git repository?

No, Just post here if you think something is missing or inaccurate.

**Comment by dajhorn** [comment]
Vice the current readme file:

1. The default directive files for wlink are renamed in recent Open Watcom releases:
```
$WATCOM/binl/wlink.lnk
$WATCOM/binl64/wlink.lnk
$WATCOM/binnt/wlink.lnk
$WATCOM/binnt64/wlink.lnk
$WATCOM/binp/wlink.lnk
$WATCOM/binw/wlink.lnk
$WATCOM/rdos/wlink.lnk
```
And they're restricted like Internet links by Explorer, which makes them slightly difficult to edit if the user is blindly following instructions on a Windows host.

2. The `binw` IDE will squawk and the `binw` WCD will crash on modern hosts.  The `binnt64` and `binl64` toolchains should be recommended for most builds.

3. The precompiled `CSTRTDHX.OBJ` has symbols that conflict with and redefine the default libraries bundled with Open Watcom v2.

I will send a patch if I get my Open Watcom v2 installation reliably integrated with HX.

**Comment by Baron-von-Riedesel** [comment]
> The precompiled CSTRTDHX.OBJ has symbols that conflict with and redefine the default libraries bundled with Open Watcom v2.

There's already a OW 2.0 version of this file - see https://github.com/Baron-von-Riedesel/HX/tree/master/Src/OWSUPP/src_20

**Comment by dajhorn** [comment]
I have all of 5da07a0b33724a7f02f9f641e600da376f984678 compiled locally and installed on my Windows 11 build host, but I'm getting the same class of glitches in OW 2.0 (64-bit, 2024-09-11)

![image](https://github.com/user-attachments/assets/31d4c708-57a3-417f-906f-3586ee4bcf99)

The application is 7-Zip 24.08 patched down to the win32c level and compiled as "system nt" such that it runs on Windows 95.

`HXLDR32.EXE` handles it perfectly, but I want to compile it as "system hxnts" so that I can package a static binary for DOS.  Would you suggest how I can troubleshoot this?

Right now I'm getting `Exception 0E` faults like this, even with a large `option stack=524288`:

![command_001](https://github.com/user-attachments/assets/bf0d4f17-8f09-4205-8024-708fc1d8ce5c)

**Comment by Baron-von-Riedesel** [comment]
> but I'm getting the same class of glitches in OW 2.0 (64-bit, 2024-09-11)

to get rid of those warnings the IMPHLP.LIB library should be added before dkrnl32s.lib. Right now I can't remember why that's necessary.

> Right now I'm getting Exception 0E faults like this, even with a large option stack=524288:

the kernel32 emulation code must be initialized. In the dkrnl32.dll it's done automatically when it's loaded. If that code is linked statically, this has to be done on startup. That's what file initw3OW.obj does - it must be added to the wlink cmdline with "Libfile initw3ow.obj".

File OWwin32.mak of the JWasm project may serve as an example how the Win32 emulation code is statically linked to binaries created with OW.

**Comment by dajhorn** [comment]
The hint is appreciated. Thanks.

For posterity, I got it working with these wlink directives:

```
system begin hxnts
  option osname='HX Extender (DOS PE, static Win32)'
  disable 1027,1030,1171
  libpath %WATCOM%/lib386
  libpath %WATCOM%/lib386/nt
  libpath /hx/lib
  libfile initw3ow.obj
  library imphlp,dkrnl32s,duser32s,ole32s,oleautos
  option stack=0x80000,stub=/hx/bin/hdld32.bin
  format windows nt ^
  runtime console=4.0
end
```

Plus the `Eliminate dead code [op el]` linker option must be disabled or parts of HX will be stripped and the EXE will crash at runtime with unusual exceptions and/or stack dumps.
